src/main/java/rx/subjects/SubjectSubscriptionManager.java

```diff
@@ -35,9 +35,7 @@
 @SuppressWarnings({"unchecked", "rawtypes"})
 /* package */final class SubjectSubscriptionManager<T> implements OnSubscribe<T> {
     /** Contains the unsubscription flag and the array of active subscribers. */
-    volatile State<T> state = State.EMPTY;
-    static final AtomicReferenceFieldUpdater<SubjectSubscriptionManager, State> STATE_UPDATER
-            = AtomicReferenceFieldUpdater.newUpdater(SubjectSubscriptionManager.class, State.class, "state");
+    AtomicReference<State<T>> state = new AtomicReference<State<T>>(State.EMPTY);
```

Comment:
Extending `AtomicReference` here conflicts with the existing `set` and `get` methods. Those methods apply to `latest`, not to `state`. We'd have to rename those if we wanted to use that pattern here